DateTimeCalendar/App.java

```diff
@@ -0,0 +1,605 @@
+        unitTestDateTimeSubtract(ShowTestOutput.ON_ERROR);
+        unitTestDateTimeDiff(ShowTestOutput.NONE);
+
+        DateTime myNewDateTime1 = new DateTime(2019, Date.Month.NOV, 6, 8, 15, 0);
+        DateTime myNewDateTime2 = new DateTime(2019, Date.Month.NOV, 6, 16, 40, 0);
+        DateTime myNewDateCompare = myNewDateTime1.diff(myNewDateTime2);
+        System.out.println(String.format("%d days, %d:h %d:m %d:s", myNewDateCompare.getDate().getDayspan(), myNewDateCompare.getTime().getHour(), myNewDateCompare.getTime().getMinute(false), myNewDateCompare.getTime().getSecond(false) ));
+
+        unitTestCalendarCalcDayOfWeek(ShowTestOutput.ON_ERROR);
+
+        DateTime[] dta = new DateTime[3];
+        dta[0] = new DateTime(1900, Date.Month.JAN, 1, 12, 30, 30);
+        dta[1] = new DateTime(1900, Date.Month.JAN, 1, 12, 30, 30);
+        dta[2] = new DateTime(1900, Date.Month.JAN, 1, 12, 30, 29);
+
+        genericTestMethod(dta, myNewDateTime2);
+
+        Date dt11 = new Date(1900, Date.Month.FEB, 1);
+        Date dt12 = dt11.subtract(3000, 1, 1);
+        System.out.println(dt2.toString());
+        int testInt = 1;
+        int testInt1 = 4;
+        System.out.println(testInt -= testInt1);
+    }
+
+    private static void unitTestDateCompareTo(Date obj1, Date obj2, int expectedOut){
+        int result = obj1.compareTo(obj2);
+        boolean testPass = false;
+
+        if (result == expectedOut){
+            testPass = true;
+        }
+
+        System.out.println(String.format("Test Date.compareTo() obj1:%s obj2:%s  ExpectedOut:%d TestPass:%s",obj1.toString(), obj2.toString(), result, testPass ));
+    }
+
+    private static void unitTestDateDiff(ShowTestOutput selectedTypeOfOutput){
+        //Wondering how to test for exception, and know that's the result I want
+        //showOutput//0 = Show nothing, 1 = show failures, 2 = show all
+        //expectedResult// -1 = exception thrown
+        Date[] testCases = new Date[10]; 
+        testCases[0] = new Date(1900, Date.Month.JAN, 0);
+        testCases[1] = new Date(1900, Date.Month.JAN, 0);
+        testCases[2] = new Date(1900, Date.Month.JAN, 1);
+        testCases[3] = new Date(1900, Date.Month.JAN, -1);
+        testCases[4] = new Date(0, Date.Month.JAN, 0);
+        testCases[5] = new Date(1900, Date.Month.JAN, 31);
+        testCases[6] = new Date(1900, Date.Month.FEB, 28);
+        testCases[7] = new Date(1900, Date.Month.AUG, 15);
+        testCases[8] = new Date(1900, Date.Month.DEC, 31);
+        testCases[9] = new Date(1904, Date.Month.FEB, 29);
+
+        int[] expectedResult = new int[10];
+        expectedResult[0] = 0;
+        expectedResult[1] = 0;
+        expectedResult[2] = 1;
+        expectedResult[3] = 1;
+        expectedResult[4] = -1;
+        expectedResult[5] = 31;
+        expectedResult[6] = 59;
+        expectedResult[7] = 227;
+        expectedResult[8] = 365;
+        expectedResult[9] = 1520;
```

Comment:
Why on this one you didn't use a helper class?

Reply:
Other than forgetting it, no. I have added it/tested it, and it will be in the next push.